app/src/marketplace/utils/smartContract.js

```diff
@@ -141,7 +141,7 @@ export const deploy = (contract: SmartContractMetadata, args: Array<any>, option
                 })
                 .on('error', errorHandler)
                 .on('transactionHash', (hash) => {
-                    emitter.emit('transactionHash', hash)
+                    emitter.emit('transactionHash', hash, futureAddress)
```

Comment:
@tumppi Changed the calculated address to be given here along with the hash once the transaction starts.